### PR TITLE
Update buildpack

### PR DESCRIPTION
**Pull request by Sveder**

### CODE REVIEW

**Comment by Sveder** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;O1HHzDSw&#x2F;165-is-there-an-option-to-make-heroku-auto-run-migrations-so-that-we-don-t-have-to-remember-for-example-they-weren-t-ran-on-marketri"> Is there an option to make heroku auto run migrations so that we don&#39;t have to remember (for example they weren&#39;t ran on marketricks)</a>

File: bin/steps/cryptography [hunk 1]
```diff
@@ -20,18 +20,16 @@ source $BIN_DIR/utils
 bpwatch start libffi_install
 
 # If a package using cffi exists within requirements, use vendored libffi.
-if (pip-grep -s requirements.txt bcrypt cffi cryptography django[bcrypt] Django[bcrypt] pyOpenSSL PyOpenSSL requests[security] &> /dev/null) then
+if (pip-grep -s requirements.txt bcrypt cffi cryptography django[bcrypt] Django[bcrypt] PyNaCl pyOpenSSL PyOpenSSL requests[security] misaka &> /dev/null) then
 
-  if [ -d ".heroku/vendor/lib/libffi-3.1.1" ]; then
-    export LIBFFI=$(pwd)/vendor
-  else
+  if [ ! -d ".heroku/vendor/lib/libffi-3.1" ]; then
```
Comment (boarik) [inline bin/steps/cryptography:25]: Is the change from 3.1.1 to 3.1 a typo?

File: builds/runtimes/pypy-5.1.0 [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
```
Comment (boarik) [inline builds/runtimes/pypy-5.1.0:1]: If this can be made into a util function, please do.

**Comment by boarik** [comment]
Round 1 done.

**Comment by Sveder** [comment]
@boarik I wrote 0% of this code. This is a merge from the head of https://github.com/heroku/heroku-buildpack-python to update our buildpack.

File: builds/runtimes/pypy-5.1.0 (hunk 2 above)
Comment (Sveder) [reply builds/runtimes/pypy-5.1.0:1]: What do you mean by this?

File: bin/steps/cryptography (hunk 1 above)
Comment (Sveder) [reply bin/steps/cryptography:25]: Feel free to understand yourself:
https://github.com/Lightricks/heroku-buildpack-python-openssl-1.0.2/commit/5438aab12146a9a1addce263789fd2cd62df1de7

Comment (boarik) [reply bin/steps/cryptography:25]: You could just say no...

File: builds/runtimes/pypy-5.1.0 (hunk 2 above)
Comment (boarik) [reply builds/runtimes/pypy-5.1.0:1]: The code in this file seems to repeat itself in several other files with only the parameters changing.

**Comment by boarik** [comment]
You could have write a description for the PR stating that now couldn't you?
Have you consulted Matan on this? He made changes to the previous buildback, which I don't recall.

File: bin/steps/cryptography (hunk 1 above)
Comment (Sveder) [reply bin/steps/cryptography:25]: Well, I researched that it is not, so I thought you'd appreciate the explanation, not only no (I know I would).

File: builds/runtimes/pypy-5.1.0 (hunk 2 above)
Comment (Sveder) [reply builds/runtimes/pypy-5.1.0:1]: Oh, well, I guess we should send a patch to heroku about this, not just change this in our repo.

**Pull request merged**